lib/services/publish-service.ts

```diff
@@ -0,0 +1,211 @@
+        return (() => {
+            var ftpPublishConnection: IPublishConnection = {
+                type: PublishService.ALLOWED_COONECTION_TYPE,
+                publicUrl: PublishService.PUBLIC_URL_DEFAULT_VALUE,
+                publishUrl: publishUrl,
+                name: name
+            };
+
+            if (!ftpPublishConnection.name) {
+                ftpPublishConnection.name = this.$prompter.getString("Name:").wait();
+            }
+
+            if (!ftpPublishConnection.publishUrl) {
+                ftpPublishConnection.publishUrl = this.$prompter.getString("Publish URL:").wait();
+            }
+
+            this.allPublishConnections.push(ftpPublishConnection);
+            this.savePublishJsonFile();
+            this.$logger.info("Connection '%s' added successfully.", ftpPublishConnection.name);
+        }).future<void>()();
+    }
+
+    public removeConnection(idOrName: string): IFuture<void> {
+        return (() => {
+            var id = parseInt(idOrName);
+            var message : string;
+            if (!isNaN(id)) {
+                this.validateId(id);
+
+                // (id - 1) is the index of the object to be deleted
+                _.remove(this.allPublishConnections, this.ftpPublishConnections[id - 1]);
+                message = util.format("Connection '%s' deleted successfully", this.ftpPublishConnections[id - 1].name);
+            } else {
+                var name = idOrName;
+                var connectionsToBeRemoved = _.filter(this.allPublishConnections, function(connection) {
+                        return connection.name === name;
+                });
+
+                if (connectionsToBeRemoved.length > 1) {
+                    this.$logger.info("Multiple connections with name '%s' found." + os.EOL + 
+                                        "Type in the index of the one you want to delete " + os.EOL + 
+                                        "Or 0 if you want to delete them all", name);
+
+                    this.listConnections(connectionsToBeRemoved);
+                    var ftpConnectionId = parseInt(this.$prompter.getString("Index:").wait());
+                    if (isNaN(ftpConnectionId) || ftpConnectionId > connectionsToBeRemoved.length || ftpConnectionId < 0) {
+                        this.$errors.failWithoutHelp("Invalid index");
+                    }
+
+                    if (ftpConnectionId === 0) {
+                        _.remove(this.allPublishConnections, function(connection) {
+                                return connection.name === name;
+                            });
+                        message = util.format("Connections named '%s' deleted successfully", name);
+                    } else {
+                        // again (id - 1) is the index of the object to be deleted
+                        _.remove(this.allPublishConnections, connectionsToBeRemoved[ftpConnectionId - 1]);
```

Comment:
I think remove operation is not correct here. First of all it doesn't modify the original array, so you should add `this.allPublishConnection = ` at the beginning. Also Remove operation with object is a little bit tricky, for example consider the following script:
```JavaScript
var b = ["a", "b", "c", "a"];
var a = _.remove(b, b[0])
console.log("b is: " + b);
console.log("a is " + a);
```

The output is:
```
b is: a,b,c,a
a is
```